.github/workflows/psalm.yml

```diff
@@ -19,12 +19,30 @@ jobs:
     name: Psalm
     runs-on: Ubuntu-20.04
 
+    env:
+      php-version: '8.1'
+      extensions: json,couchbase,memcached,mongodb,redis,xsl,ldap,dom
     steps:
+      - name: Setup cache environment
+        id: extcache
+        uses: shivammathur/cache-extensions@v1
+        with:
+          php-version: ${{ env.php-version }}
+          extensions: ${{ env.extensions }}
+          key: cache-v1 # can be any string, change to clear the extension cache.
```

Comment:
on 6.3 the list of extension is not the same as on 6.2/5.4
shouldn't we vary this by php-version + extensions?

Reply:
I haven't used GitHub Actions caching much, but I think it will [attempt to restore the cache from the current branch before falling back to the default branch](https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#matching-a-cache-key). So there shouldn't be any harm in reusing a cache key across branches.